core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -386,83 +416,67 @@ func getChainReaderConfig(
 	if chainID == homeChainID {
 		chainReaderConfig = evmconfig.MergeReaderConfigs(chainReaderConfig, evmconfig.HomeChainReaderConfigRaw)
 	}
-	return chainReaderConfig
-}
 
-func createChainReader(
-	lggr logger.Logger,
-	chain legacyevm.Chain,
-	chainReaderConfig evmrelaytypes.ChainReaderConfig,
-	pluginType cctypes.PluginType,
-) (types.ContractReader, error) {
-	cr, err := evm.NewChainReaderService(
-		context.Background(),
-		lggr.
-			Named("EVMChainReaderService").
-			Named(chain.ID().String()).
-			Named(pluginType.String()),
-		chain.LogPoller(),
-		chain.HeadTracker(),
-		chain.Client(),
-		chainReaderConfig,
-	)
+	marshaledConfig, err := json.Marshal(chainReaderConfig)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create contract reader for chain %s: %w", chain.ID(), err)
+		return nil, fmt.Errorf("failed to marshal chain reader config: %w", err)
 	}
-	return cr, nil
-}
 
-func bindContracts(
-	chain legacyevm.Chain,
-	cr types.ContractReader,
-	config cctypes.OCR3ConfigWithMeta,
-	destChainID uint64,
-) error {
-	if chain.ID().Uint64() == destChainID {
-		offrampAddressHex := common.BytesToAddress(config.Config.OfframpAddress).Hex()
-		err := cr.Bind(context.Background(), []types.BoundContract{
-			{
-				Address: offrampAddressHex,
-				Name:    consts.ContractNameOffRamp,
-			},
-		})
-		if err != nil {
-			return fmt.Errorf("failed to bind chain reader for dest chain %s's offramp at %s: %w", chain.ID(), offrampAddressHex, err)
-		}
-	}
-	return nil
+	return marshaledConfig, nil
 }
 
 func createChainWriter(
-	lggr logger.Logger,
-	chain legacyevm.Chain,
-	pluginType cctypes.PluginType,
+	chainID *big.Int,
+	evmConfigs toml.EVMConfigs,
+	relayer loop.Relayer,
 	transmitters map[types.RelayID][]string,
 	execBatchGasLimit uint64,
 ) (types.ChainWriter, error) {
 	var fromAddress common.Address
-	transmitter, ok := transmitters[types.NewRelayID(relay.NetworkEVM, chain.ID().String())]
+	transmitter, ok := transmitters[types.NewRelayID(relay.NetworkEVM, chainID.String())]
 	if ok {
 		// TODO: remove EVM-specific stuff
 		fromAddress = common.HexToAddress(transmitter[0])
 	}
-	cw, err := evm.NewChainWriterService(
-		lggr.Named("EVMChainWriterService").
-			Named(chain.ID().String()).
-			Named(pluginType.String()),
-		chain.Client(),
-		chain.TxManager(),
-		chain.GasEstimator(),
-		evmconfig.ChainWriterConfigRaw(
-			fromAddress,
-			chain.Config().EVM().GasEstimator().PriceMaxKey(fromAddress),
-			defaultCommitGasLimit,
-			execBatchGasLimit,
-		),
+
+	var maxGasPrice *assets.Wei
+	for _, config := range evmConfigs {
+		if config.ChainID.ToInt().Cmp(chainID) == 0 {
```

Comment:
nit, but I think if you reverse if stmt it would be less indents in the code and would make it easier to read

```go
if config.ChainID.ToInt().Cmp(chainID) != 0 {
      continue
}
// current logic goes here
```